src/java.base/share/classes/jdk/internal/misc/CDS.java

```diff
@@ -73,9 +73,9 @@ public static boolean isDumpingClassList() {
     public static native long getRandomSeedForDumping();
 
     /**
-     * Check if dynamic dumping is enabled via the DynamicDumpSharedSpaces flag.
+     * Check if CDS dumping is enabled via the DynamicDumpSharedSpaces or the DumpSharedSpaces flag.
      */
-    public static native boolean isDynamicDumpingEnabled(); // will return false for static dumping.
+    public static native boolean isDumpingEnabled();
```

Comment:
I think it will be more consistent if we use the same pattern as `CDS::isDumpingClassList()`

```
    private static final boolean isDumpingArchive;
    static {
        isDumpingClassList = isDumpingArchive0();
    }

    /**
      * Is the VM writing to a (static or dynamic) CDS archive.
      */
    public static boolean isDumpingArchive() {
        return isDumpingArchive;
    }
```

Then in LambdaProxyClassArchive.java, there's no need to keep a separate variable of dumpArchive. You can simply call CDS.isDumpingArchive(). The JIT compiler will automatically inline the call so it will be just as fast.

LambdaProxyClassArchive::sharingEnabled should also be rewritten to use this pattern.

Reply:
I've changed the API's in CDS.java to be consistent with CDS.isDumpingClasslist.